folio-spring-i18n/src/main/java/org/folio/spring/i18n/service/TranslationService.java

```diff
@@ -0,0 +1,352 @@
+ * serve the user the best possible translation.
+ */
+@Log4j2
+@Service
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
+public class TranslationService {
+
+  /**
+   * The location of all translations in the classpath. Allows for format matching UI modules:
+   *  /translations/{module-name}/{locale}.json
+   */
+  private static final String TRANSLATIONS_CLASSPATH = "classpath:%s*/*.json";
+
+  /**
+   * A map from language code to a map from country to the JSON resource containing {key to ICU format string mappings}.
+   * Will be populated the first time any translation is requested.
+   * Note that files are not read here, only enumerated.  Look at the other maps for that information
+   *
+   * <p>This can be null - if this is the case, the map has not been constructed yet.</p>
+   * <p>You should call getFileMap instead of accessing this directly.</p>
+   *
+   * <p>
+   *  Language code ({@link TranslationFile#UNKNOWN_PART TranslationFile#UNKNOWN_PART}) will be used if unknown
+   * </p>
+   */
+  @Nullable
+  protected Map<String, Map<String, TranslationFile>> translationFileFromLanguageCountryMap =
+    null;
+
+  /**
+   * A map from locales to translations, filled in on-demand as locales are presented.
+   */
+  protected Map<Locale, TranslationMap> localeTranslations = new HashMap<>();
+
+  private final ResourcePatternResolver resourceResolver;
+  private final TranslationConfiguration configuration;
+
+  /**
+   * Get a map from language -&gt; country -&gt; {@link TranslationFile TranslationFile}.
+   * The files are pulled from the filesystem based on the provided
+   * {@link TranslationConfiguration TranslationConfiguration}.
+   *
+   * @return the map of languages to countries to {@link TranslationFile TranslationFile}s
+   */
+  @Nonnull
+  public Map<String, Map<String, TranslationFile>> getFileMap() {
+    if (this.translationFileFromLanguageCountryMap == null) {
+      this.translationFileFromLanguageCountryMap =
+        buildLanguageCountryPatternMap();
+    }
+    return this.translationFileFromLanguageCountryMap;
+  }
+
+  /**
+   * Get the best {@link TranslationMap TranslationMap} associated with the current locale.
+   *
+   * @param locale the locale to find a TranslationMap for
+   * @param fallback the default translation that should be used as a fallback; can be null when
+   *   creating the primary fallback translation
+   * @return the best translation available for the current locale, potentially null if the
+   *   default is null.  It checks for match of language and country, then language, then returns
+   *   default if neither are available
+   */
+  @Nullable
+  public TranslationMap getTranslation(
+    Locale locale,
+    @Nullable TranslationMap fallback
+  ) {
+    return localeTranslations.computeIfAbsent(
+      locale,
+      (Locale missingLocale) -> {
+        log.info("Cache miss on {}; loading map", missingLocale);
+
+        if (
+          this.getFileMap()
```

Comment:
Why do all inner methods are used with `this`? Are there any pros?

Reply:
I prefer to use it; I think it makes it more readable and removes ambiguity